My idea: Fully worked out

MrKickkiller opened the pull request:
PlaneFactory: Still bears my irl name. Suggest change in later commit.

MrKickkiller committed:
My Idea: worked out

Simply calling PatternFactorie.create<Shape>(param) will return a
pattern with that shape and according to params :panda_face:

MrKickkiller committed:
Create factories to make patterns. Changed the pattern to an object instead of a cubic char array

Possible: Intersect 2 patterns

MrKickkiller committed:
Added combining of 2 patterns.
The pattern on which the operation is called will be considered as the primary one. AKA it's values WILL OVERRIDE the other's value when the primary value != ''

AEnterprise commented:
hmm, not sure if we realy want this. seems to make things a bit overcomplex for what we are making atm

AEnterprise commented on src/main/java/kineticrevolution/multiblocks/patterns/PatternFactories/Factories/PlaneFactory.java:
```diff
@@ -0,0 +1,21 @@
+import kineticrevolution.multiblocks.patterns.Pattern;
+
+/**
+ * Created by Mathieu on 3/07/2015.
```
might wana fix the name here

AEnterprise commented on src/main/java/kineticrevolution/multiblocks/patterns/PatternFactories/Factories/PlaneFactory.java:
```diff
@@ -0,0 +1,21 @@
+public class PlaneFactory implements IPatternFactory {
+    @Override
+    public Pattern createPattern(int size, char a) throws PatternException {
+        char[][][] pattern = new char[size][size][1];
```
this only allows squares, could you make it allow for rectangles as well?

MrKickkiller committed:
Made it so that there is compatibality/ room to create eg HollowBeam and such.
Translation implemented (only works with positive delta's)
A pattern is now final too.

MrKickkiller committed:
Changed the parameters of the methods in PatternFactory. Left unchanged would not compile

MrKickkiller committed:
Beam and HollowBeam factory done.

AEnterprise commented on src/main/java/kineticrevolution/multiblocks/patterns/PatternFactories/PatternFactory.java:
```diff
@@ -0,0 +1,53 @@
+ * Created by MrKickkiller on 3/07/2015.
+ */
+public final class PatternFactory {
+    final static ArrayList<IPatternFactory> factories = new ArrayList<>();
```
a list and numbered factories doesn't seem the best approach to me, i think an enum would be better here

AEnterprise commented:
think we'll have this and the current way of doing it manually to give people the choise of what to use. could you provide a few examples on how exactly it works as well?

MrKickkiller commented:
Example:
Pattern patt = PatternFactory.createHollowBeam(character to be filled in, list of 3 ints, indicating x, y, z)
<patt will now be a Pattern that has a field of char[][][]>

Pattern patt2 = patt.translation(x,y,z) will result in the return a new Pattern with all elements translated x, y, z. Xyz are all positive. no support for negative delta's.

patt.combine(patt2) will result in a new Pattern that has the 2 patterns merged. patt's chars are prefered over patt's

AEnterprise commented:
merged
